rest_framework/serializers.py

```diff
@@ -1131,9 +1131,14 @@ def build_relational_field(self, field_name, relation_info):
         field_kwargs = get_relation_kwargs(field_name, relation_info)
 
         to_field = field_kwargs.pop('to_field', None)
-        if to_field and not relation_info.related_model._meta.get_field(to_field).primary_key:
-            field_kwargs['slug_field'] = to_field
-            field_class = self.serializer_related_to_field
+        if relation_info.reverse:
+            if to_field and not relation_info.related_model_field.related_fields[0][1].primary_key:
```

Comment:
I will need to go back in and check what the actual values are.  I really should have put a comment in there when I made the change.  I'm pretty sure it's grabbing the ```to_field``` value for the related field (see the relevant django code [here](https://github.com/django/django/blob/a4c20ae85b40c49e28d1b2227208e4f00d7820df/django/db/models/fields/related.py#L563)) but you are totally right, it is very hard to decipher just by looking at it.